allennlp/modules/time_distributed.py

```diff
@@ -0,0 +1,36 @@
+import torch
+
+
+class TimeDistributed(torch.nn.Module):
+    """
+    Given an input shaped like ``(batch_size, time_steps, [rest])`` and a ``Module`` that takes
+    inputs like ``(batch_size, [rest])``, ``TimeDistributed`` reshapes the input to be
+    ``(batch_size * time_steps, [rest])``, applies the contained ``Module``, then reshapes it back.
+
+    Note that while the above gives shapes with ``batch_size`` first, this ``Module`` also works if
+    ``batch_size`` is second - we always just combine the first two dimensions, then split them.
+
+    This behaves identically to Keras' TimeDistributed layer.
```

Comment:
Maybe remove this - no need to compare Keras, particularly as 1) Keras doesn't support having batch size second and 2) this doesn't have any concept of masking yet.